### ISSUE TITLE
onDelete property on foreign key is not modified on sync

### ISSUE THREAD

**Issue by Ionaru** [typeorm/typeorm#1261]
**Reproduction code**
```typescript
import { BaseEntity, createConnection, Entity, JoinColumn, OneToOne, PrimaryGeneratedColumn } from 'typeorm';
import 'reflect-metadata';

@Entity()
class Foo extends BaseEntity {

    @PrimaryGeneratedColumn()
    public id: number;
}

@Entity()
class Bar extends BaseEntity {

    @PrimaryGeneratedColumn()
    public id: number;

    @OneToOne(type => Foo, {
        onDelete: 'RESTRICT'
    })
    @JoinColumn()
    foo: Foo;
}

async function run() {
    await createConnection({
        type: 'sqlite',
        database: 'test.db',
        entities: [
            Foo, Bar
        ],
        logging: 'all',
        synchronize: true,
    });

    const foo = new Foo();
    const bar = new Bar();
    bar.foo = foo;
    await foo.save();
    await bar.save();
    await foo.remove();
}

run().then();
```
**Run steps**
1. Run the program once, it will error because of a foreign key constraint when deleting `foo` (correct).
2. For the property `onDelete` change `RESTRICT` to `SET NULL` or `CASCADE`
3. Run the program again, it will error because of a foreign key constraint when deleting `foo`. (wrong)

**Info**
We changed the foreign key `onDelete` property, but this change was not saved to the database. This resulted in a foreign key constraint error.

I don't know if this is the only property that is not saved, there might be others.

**Comment by pleerock** [comment]
Its a bug. @AlexMesser working currently on schema synchronization refactoring and we'll ship all new changes and this fix as well in 0.2.0

Also please take a look how I added test for your issue, you can same next time ;)

**Comment by stonecold123** [comment]
When the table already exists, I'm not able to add/update the onDelete property as well for the relationship.
I have to add it manually or delete the entire table to add the onDelete property.

**Comment by AlexMesser** [comment]
fixed and released in `typeorm@0.2.0-alpha.23`. You can try it via `npm i typeorm@next`.
